### PR TITLE
Adding a roadmap.

### PR DESCRIPTION
**Pull request by vishh**
Updated readme to include a link to kubedash.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Roadmap
+
+## Heapster v0.19
```
Comment (dchen1107) [inline docs/roadmap.md]: timeline? Kubernetes 1.2?

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+- Support for InfluxDB v0.9
+
+## Heapster v0.20
```
Comment (dchen1107) [inline docs/roadmap.md]: timeline?

File: docs/roadmap.md (hunk 1 above)
Comment (vishh) [inline docs/roadmap.md]: Updated with tentative timelines.

File: docs/roadmap.md (hunk 2 above)
Comment (vishh) [inline docs/roadmap.md]: Done

**Comment by vishh** [comment]
cc @piosz @mwielgus

**Comment by piosz** [comment]
cc @jszczepkowski @fgrzadkowski

**Comment by vishh** [comment]
@dchen1107: Ping

**Comment by fgrzadkowski** [comment]
@vishh Who will be working on those new features in heapster?

**Comment by vishh** [comment]
@fgrzadkowski: That is TBD. I intend to handle some of the community work that are not specific to k8s.

**Comment by fgrzadkowski** [comment]
@vishh @dchen1107 Let's sync on Thu about this. I think it's important who owns this so that we all properly plan work for 1.2.

**Comment by liuhewei** [comment]
Interesting features! Are there already some introduction or draft design docs to be shared? Thanks.

**Comment by vishh** [comment]
@liuhewei: Is there a specific feature that you are interested in? Some of the deliverables in the roadmap are yet to be scoped out.

**Comment by vishh** [comment]
I have updated the roadmap based on your offline sync up.
PTAL @jszczepkowski @wojtek-t @piosz @dchen1107

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### Timeline
+December 2015
+
+### Features
+- Support for application metrics.
+- Improving existing model API
+  - Support label based queries in the APIs
+- Internal optimizations for performance and stability
+- Scalability - support <= 250 node clusters
```
Comment (davidopp) [inline docs/roadmap.md]: Maybe just "support 250 node clusters" instead of \<= so people know we will actually support 250.

Same for \<= 1000 later.

**Comment by dchen1107** [comment]
LGTM thanks!

**Comment by liuhewei** [comment]
@vishh From the short-term feature list, we're very interested in the "Support for application metrics". Actually we're also thinking about how to collect more application-level metrics inside containers. I assume this feature is related to cAdvisor's design [here](https://github.com/google/cadvisor/blob/master/docs/application_metrics.md)? And, the scalability and performance are also key points. We're preparing to do some stress testing on our deployment later. By the way, we'll submit our kafka-sink as a PR around this weekend for review.

As for the long-term and future features, these policy-related features such as "Resource usage predictions, events generating and outlier detection" are also interesting. Actually we're doing some researches on these directions. We have our own policy-engine to analyze the data collected from heapster. If heapster can do those things, we're happy to follow the progress.

And I see an ecosystem related feature: Support for Docker Swarm and Mesos. I wonder if this feature need cAdvisor also be installed on Mesos or Swarm? It will be great if heapster's source-configuration is flexible enough to connect to other container management systems. 

We are also thinking about some other features not in the roadmap, for example, multiple heapster instances for HA, one heapster instance for multiple K8S clusters or partial nodes of one K8S cluster (for multi-tenant scenarios). 

cc @alfred-huangjian @huangyuqi

**Comment by vishh** [comment]
@liuhewei: Thanks for explaining.
>  From the short-term feature list, we're very interested in the "Support for application metrics". Actually we're also thinking about how to collect more application-level metrics inside containers. I assume this feature is related to cAdvisor's design here?
Yes. It will be based on cAdvisor's app metrics design. AFAIK, there are no better alternatives to cAdvisor when it comes to container workloads for this purpose.

>And, the scalability and performance are also key points. We're preparing to do some stress testing on our deployment later. 
That's great. @piosz is also looking into scalability testing of heapster. You can possibly collaborate with him. 

> By the way, we'll submit our kafka-sink as a PR around this weekend for review.
That's great. Looking forward to it!

> As for the long-term and future features, these policy-related features such as "Resource usage predictions, events generating and outlier detection" are also interesting. Actually we're doing some researches on these directions. We have our own policy-engine to analyze the data collected from heapster. If heapster can do those things, we're happy to follow the progress.

We'd love to collaborate in this area. Contributions in the form of design or PRs are welcome :)

> And I see an ecosystem related feature: Support for Docker Swarm and Mesos. I wonder if this feature need cAdvisor also be installed on Mesos or Swarm? It will be great if heapster's source-configuration is flexible enough to connect to other container management systems.

This has been something that I intend to work on soon. I believe heapster is generic enough to support any container cluster. It can already support CoreOS clusters. I plan on adding a consul discovery plugin to support swarm clusters. #136 has been open for a while as well.

> We are also thinking about some other features not in the roadmap, for example, multiple heapster instances for HA, one heapster instance for multiple K8S clusters or partial nodes of one K8S cluster (for multi-tenant scenarios).

We have talked briefly about improving reliability by replicating heapster. It should be possible today to run multiple instances of heapster for HA purposes. If we shard heapster and make it self-healing, replication might not be necessary. 

Running a single heapster for multiple cluster's is an interesting idea. ,Heapster will be used for resource metrics purposes in Kubernetes. I'm not sure how we will handle cluster identity and auth if we were to run a single heapster for multiple clusters.

File: docs/roadmap.md (hunk 3 above)
Comment (vishh) [inline docs/roadmap.md]: Makes sense. Updated.

**Pull request merged**